coriolis/providers/base.py

```diff
@@ -34,13 +34,15 @@ class BaseEndpointInstancesProvider(BaseEndpointProvider):
     """Defines operations for listing instances off of Endpoints."""
 
     @abc.abstractmethod
-    def get_instances(self, ctxt, connection_info, limit=None,
-                      last_seen_id=None, instance_name_pattern=None):
+    def get_instances(self, ctxt, connection_info, source_environment,
```

Comment:
This will require that all providers be updated. Is source_environment necessary for all providers?

Reply:
Not all, but some clouds which have various non-connection-related params (ex: resgroup on Azure or AZ on Amazon), and IMO it's better to make it part of the interface for all plugins IMO moving forward.
Additionally, this will allow for the updating of other plugins which already implement these various params as part of the endpoint connection info (ex: the various region params in OpenStack endpoints)